fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/LAORequestFactory.java

```diff
@@ -0,0 +1,31 @@
+package com.github.dedis.popstellar.repository.remote;
+
+import androidx.annotation.NonNull;
+
+import com.tinder.scarlet.websocket.okhttp.request.RequestFactory;
+
+import okhttp3.Request;
+
+public class LAORequestFactory implements RequestFactory {
+
+  @NonNull private static String url = "ws://10.0.2.2:9000/organizer/client";
+
+  public LAORequestFactory(@NonNull String defaultUrl) {
+    setUrl(defaultUrl);
+  }
+
+  @NonNull
+  @Override
+  public Request createRequest() {
+    return new Request.Builder().url(url).build();
+  }
+
+  @NonNull
+  public static String getUrl() {
+    return url;
+  }
+
+  public static void setUrl(@NonNull String newUrl) {
+    url = newUrl;
+  }
```

Comment:
You are not using the object-oriented in the right way.

Making the url field static prevent the use of multiple LAORequestFactory as they will all share the same url.

The idea is also that the default value is given as argument.

```suggestion
  @NonNull private String url;

  public LAORequestFactory(@NonNull String defaultUrl) {
    url  = defaultUrl;
  }

  @NonNull
  @Override
  public Request createRequest() {
    return new Request.Builder().url(url).build();
  }

  @NonNull
  public String getUrl() {
    return url;
  }

  public void setUrl(@NonNull String newUrl) {
    url = newUrl;
  }
```